src/main/java/com/simprints/simface/quality/MlKitFaceDetectionProcessor.kt

```diff
@@ -136,4 +146,286 @@ internal class MlKitFaceDetectionProcessor() : FaceDetectionProcessor {
 
         return (leftEyeScore + rightEyeScore) / 2.0
     }
+
+    private fun buildLandmarks(face: Face): FacialLandmarks? {
+        val leftEye = face.getLandmark(FaceLandmark.LEFT_EYE)?.position
+            ?: face.getContour(FaceContour.LEFT_EYE)?.points?.getOrNull(4)
+            ?: return null
+
+        val rightEye = face.getLandmark(FaceLandmark.RIGHT_EYE)?.position
+            ?: face.getContour(FaceContour.RIGHT_EYE)?.points?.getOrNull(4)
+            ?: return null
+
+        val nose = face.getLandmark(FaceLandmark.NOSE_BASE)?.position
+            ?: face.getContour(FaceContour.NOSE_BRIDGE)?.takeIf { it.points.isNotEmpty() }?.points?.last()
+            ?: return null
+
+        val mouthLeft = face.getLandmark(FaceLandmark.MOUTH_LEFT)?.position
+            ?: face.getContour(FaceContour.LOWER_LIP_BOTTOM)?.takeIf { it.points.isNotEmpty() }?.points?.last()
+            ?: return null
+
+        val mouthRight = face.getLandmark(FaceLandmark.MOUTH_RIGHT)?.position
+            ?: face.getContour(FaceContour.LOWER_LIP_BOTTOM)?.takeIf { it.points.isNotEmpty() }?.points?.first()
+            ?: return null
+
+        return FacialLandmarks(
+            Point2D(leftEye.x, leftEye.y),
+            Point2D(rightEye.x, rightEye.y),
+            Point2D(nose.x, nose.y),
+            Point2D(mouthLeft.x, mouthLeft.y),
+            Point2D(mouthRight.x, mouthRight.y)
+        )
+    }
+
+
+    override fun warpAlignFace(face: FacialLandmarks, inputImage: Bitmap): Bitmap {
+        /**
```

Comment:
If this is intended as a documentation comment, it must be right before the method definition for IDEs to be able to pick it up. If it just a block comment to explain the magic numbers it should start with `/*`.